apps/anoma_lib/lib/noun/jam.ex

```diff
@@ -310,8 +310,14 @@ defmodule Noun.Jam do
                       __STACKTRACE__
           end
 
-        with {:ok, referenced_noun} <-
-               Map.fetch(cache, :binary.decode_unsigned(backref_key, :little)) do
+        backref_offset = :binary.decode_unsigned(backref_key, :little)
+
+        if backref_offset >= offset do
+          raise CueError,
+                "invalid backref at offset #{inspect(offset)}"
+        end
+
+        with {:ok, referenced_noun} <- Map.fetch(cache, backref_offset) do
```

Comment:
Another change that doesn't make sense, we can't fetch a backref greater than the offset, this raises a CueError right away